packages/firebase_database/firebase_database/lib/src/query.dart

```diff
@@ -81,6 +81,30 @@ class Query {
   /// Listens for a single value event and then stops listening.
   Future<DataSnapshot> once() async => (await onValue.first).snapshot;
 
+  /// Gets the most up-to-date result for this query.
+  Future<DataSnapshot?> get() async {
+    final result = await _database._channel.invokeMethod<Map<dynamic, dynamic>>(
+      'Query#get',
+      <String, dynamic>{
+        'app': _database.app?.name,
+        'databaseURL': _database.databaseURL,
+        'path': path,
+      },
+    );
+    if (result == null || result is! Map) {
+      throw 'firebase_database get call failed from Dart.';
+    } else if (result.containsKey('error') && result['error'] != null) {
```

Comment:
Why is the `result['error']` needed? Is there a case where we would have `{'error': null}` send?

Reply:
Form the iOS side the returned object can will contain an `error` key even when the result is successful. On Android side there would be no `error` key at all.

Reply:
But why do we need both `containsKey` and `result['error'] != null`?

Reply:
Yes, it appears iOS sends along `{'error': null}` even when successful.